src/PrestaShopBundle/Controller/Admin/Sell/Catalog/MonitoringController.php

```diff
@@ -216,91 +241,7 @@ private function identifySearchableGrid(Request $request)
 
         return [
             'grid_id' => $gridId,
-            'grid_definition' => $this->get($gridDefinition),
+            'grid_definition' => $this->kernel->getContainer()->get($gridDefinition),
```

Comment:
You shouldn't use the kernel to get the container to get a dynamic grid definition, that's kind of what we aim to avoid with this refacto But I admit this use case is tricky, usually you only need to inject one grid definition so it's easier

In this case I think you should rely on `getSubscribedServices`, you can override the method and add all these grid services in the subscribed services, then they will be known by the controller mini container but only instanciated when you call `$this->container->get($gridDefinition)`

Phpstan will likely shout about it, we have a rule to avoid using this but I'm not sure it's still relevant We can think about how this could be removed/improved later, in the meantime you can add this controller in the exception list:
https://github.com/PrestaShop/PrestaShop/blob/62ef859fa8db79a5fdada03b28d856cdde62979d/phpstan-disallowed-calls.neon#L289